Unstructured-IO/unstructured
# build(deps): split up dependencies by document type

### Summary

Closes #611. Updates the dependencies to create a slim base installation + extras for each document type. The `all-docs` extra includes the dependencies for all doc types. The `local-inference` extra still remains as an alias for `all-docs` since that's the extra many currently users use to install all of the dependencies.

As a follow on, the PDF and image requirements could be broken down further by strategy.

### Testing

- In a fresh virtual environment, run `make install-base` and `pip install -r requirements/test.txt` and `pip install -e .`. The following tests should pass:
    - `pytest test_unstructured/partition/test_text.py`
    - `pytest test_unstructured/partition/test_email.py`
    - `pytest test_unstructured/partition/test_html_partition.py`
    - `pytest test_unstructured/partition/test_text.py`
    - `pytest test_unstructured/partition/test_xml_partition.py`
- Run `make install-csv`. The following tests should pass:
    - `pytest test_unstructured/partition/test_csv.py`
    - `pytest test_unstructured/partition/test_tsv.py`
- Run `make install-docx`. The following tests should pass:
    - `pytest test_unstructured/partition/test_doc.py`
    - `pytest test_unstructured/partition/test_docx.py`
- Run `make install-odt`. The following tests should pass:
    - `pytest test_unstructured/partition/test_odt.py`
- Run `make install-pypandoc`. The following tests should pass:
    - `pytest test_unstructured/partition/test_epub.py`
    - `pytest test_unstructured/partition/test_org.py`
    - `pytest test_unstructured/partition/test_rst.py`
    - `pytest test_unstructured/partition/test_rtf.py`
- Run `make install-markdown`. The following tests should pass:
    - `pytest test_unstructured/partition/test_md.py`
- Run `make install-msg`. The following tests should pass:
    - `pytest test_unstructured/partition/test_msg.py`
- Run `make install-pdf-image`. The following tests should pass:
    - `pytest test_unstructured/partition/test_pdf.py`
    - `pytest test_unstructured/partition/test_image.py`
- Run `make install-pptx`. The following tests should pass:
    - `pytest test_unstructured/partition/test_ppt.py`
    - `pytest test_unstructured/partition/test_pptx.py`
- Run `make install-xlsx`. The following tests should pass:
    - `pytest test_unstructured/partition/test_xlsx.py`
- Run `make install-ci` and `pip install argilla weaviate-client`. `make test` should pass


Test `test_unstructured/partition/test_json.py` is currently a little weird because there are tests in there that use `partition`, making it hard to test `partition_json` isolation. I opened #993 to address that.
